Sources/SwiftIdentifier.swift

```diff
@@ -78,5 +74,19 @@ func swiftIdentifier(from string: String,
       return ""
     }
   })
-  return prefix + mappedParts.joined(separator: replacement)
+
+  let result = mappedParts.joined(separator: replacement)
+  return prefixWithUnderscoreIfNeeded(string: result, forbiddenChars: exceptions)
+}
+
+func prefixWithUnderscoreIfNeeded(string: String,
+                                  forbiddenChars exceptions: String = "") -> String {
+
+  let (head, tail) = identifierCharacterSets(exceptions: exceptions)
+
+  let chars = string.unicodeScalars
+  let firstChar = chars[chars.startIndex]
+  let prefix = !head.longCharacterIsMember(firstChar.value) && tail.longCharacterIsMember(firstChar.value) ? "_" : ""
```

Comment:
I'm not sure I remember why we have that test logic.
Shouldn't testing against the valid characters for `head` be enough here to determine if we must prefix with `_` or not? If the first character is not in the set of valid head chars, then what's the point of testing it against tail, we will prefix it with `_` anyway…

And then if that tail is indeed not needed, we should probably split the `identifierCharacterSets` function returning a tuple into 2 functions returning a single set each I guess.